web/modules/custom/joinup_core/src/EventSubscriber/RefreshCachedFieldsEventSubscriber.php

```diff
@@ -86,89 +87,98 @@ public function __construct(EntityTypeManagerInterface $entityTypeManager, TimeI
   /**
    * {@inheritdoc}
    */
-  public function refreshExpiredField(RefreshExpiredFieldEventInterface $event) {
-    // Only react to fields that store the download count or visit count.
-    if (!in_array($event->getFieldName(), [static::FIELD_NAME_DOWNLOAD_COUNT, static::FIELD_NAME_VISIT_COUNT])) {
-      return;
+  public function refreshExpiredFields(RefreshExpiredFieldsEventInterface $event) {
+    $items = $event->getExpiredItems()->getItems();
+
+    // All requests are sent by POST method to handle the amount of concurrent
+    // requests in terms of request length.
+    $this->piwikQueryFactory->getQueryFactory()->getHttpClient()->setMethod('POST');
+    $query = $this->piwikQueryFactory->getQuery('API.getBulkRequest');
+
+    $piwik_config = $this->configFactory->get('piwik.settings');
+    $piwik_reporting_api_config = $this->configFactory->get('piwik_reporting_api.settings');
+    $site_id = $piwik_config->get('site_id');
+    $token_auth = $piwik_reporting_api_config->get('token_auth');
+
+    $sub_params = [
+      'module' => 'API',
+      'idSite' => $site_id,
+      'token_auth' => $token_auth,
+    ];
```

Comment:
All this stuff should not be needed, it is handled automatically by the Piwik Reporting API module.

Reply:
These 3 properties are put into the subqueries. Each sub query needs these information separately as they are handled independently. The reporting api module is only handling the parent query in which I do not set it.